convert.go

```diff
@@ -149,7 +149,7 @@ func underlyingTimeType(val interface{}) (interface{}, bool) {
 	switch refVal.Kind() {
 	case reflect.Ptr:
 		if refVal.IsNil() {
-			return nil, false
+			return nil, true
```

Comment:
I think this would allow any type of nil, not just a `*time.Time`. Not sure if that's really a problem.

Reply:
That was a mistake, this won't stay